server/replication.go

```diff
@@ -26,3 +26,60 @@ func newReplication(cfg *ReplicationConfig) *Replication {
 func (r *Replication) GetMaxReplicas() int {
 	return int(r.cfg.MaxReplicas)
 }
+
+// GetMaxReplicaScore returns the max replica score of a store.
+func (r *Replication) GetMaxReplicaScore() int {
+	return len(r.cfg.LocationLabels)
+}
+
+// GetReplicaScore returns the replica score of the store relative to the candidates.
+func (r *Replication) GetReplicaScore(candidates []*storeInfo, store *storeInfo) int {
+	for i, key := range r.cfg.LocationLabels {
+		value := store.labelValue(key)
+		if len(value) == 0 {
+			// If the store doesn't have this label, we assume
+			// it has the same value with all candidates.
+			continue
+		}
+
+		// Reset candidates.
+		stores := candidates
+		candidates = []*storeInfo{}
+
+		// Push stores with the same label value to candidates.
+		for _, s := range stores {
+			if s.GetId() == store.GetId() {
+				continue
+			}
+			if s.labelValue(key) == value {
+				candidates = append(candidates, s)
+			}
+		}
+
+		// If no candidates, it means the label value is different from others.
+		if len(candidates) == 0 {
+			return r.GetMaxReplicaScore() - i
+		}
+	}
+	return 0
+}
```

Comment:
This algorithm is not enough in this case:
```
Node1: zone1-host1 (*)
Node2: zone1-host2 (*)
Node3: zone1-host3
Node4: zone2-host4 (*)
Node5: zone2-host5
Node6: zone3-host6 (*)
```
We have 4 replicas on Node 1, 2, 4, 6, then we want to add the 5th. Node3 and Node5 will result in the same score, but we should choose Node5 rather than Node3.